src/glade-utils.c

```diff
@@ -2,8 +2,115 @@
 
 extern PGconn *conn;
 extern GtkBuilder *builder;
+extern current_order_t *current_order;
 extern int nb_step;
 
+void init_current_order (order_t *order) {
+    GtkBox *current_order_controls = GTK_BOX(gtk_builder_get_object(builder, "current-order-controls"));
+    gtk_widget_show_all(GTK_WIDGET(current_order_controls));
+
+    current_order = (current_order_t *)malloc(sizeof(current_order_t));
+    current_order->order = get(conn, order->id);
+    current_order->total_step = current_order->order->nb_cocktails;
```

Comment:
le nombre d'étapes c'est pas forcément le nombre de cocktails. Par exemple pour faire un sex on the beach tu as plusieurs étapes où chaque étape correspond à une action

Reply:
mince je l'avais pas compris comme ça, du coup "bottle" et "total_bottle" ça correspond à quoi pour toi?

Reply:
ah oui je vois ça aurait été mieux de les appeler genre "cocktail_number" et "cocktail_total". Enfait comme ça on a le numéro de cocktail actuel (ex : le 5 ème de la commande) par rapport au nombre de cocktails de la commande.
On aurait ducoup le 5ème cocktail / 6

Et comme ça le step permet de connaître l'avancé de la commande. 
Exemple : tu as 6 cocktails et chaque cocktails se fait en 5 étapes. Je viens de terminer le cocktail 2 ducoup j'ai fait 10 étapes au total sur les 30 à faire (donc 33%). Tu vois le truc ?

Reply:
Oui je vois, donc pour résumé on aura : 

- cocktail_number : numéro du cocktail dans la liste des cocktails de la commande
- total_cocktail : nombre total de cocktail dans la commande
- step_number : numéro de l'étape (pour plus facilement connaitre l'avancée de la commande)
- total_step : nombre total d'étapes (en prenant donc en compte tous les cocktails de la commande)
- bottle : numéro de l'étape dans le cocktail qu'on est en train de faire actuellement
- total_bottle : nombre d'étapes pour le cocktail qu'on est en train de faire actuellement

ça te va ?

Reply:
nickel !